subprojects/gdk_rust/gdk_electrum/src/lib.rs

```diff
@@ -347,32 +345,22 @@ impl Session<Error> for ElectrumSession {
         )
         .ok_or(Error::InvalidMnemonic)?;
         let secp = Secp256k1::new();
-        let xprv =
+        let master_xprv =
             ExtendedPrivKey::new_master(bitcoin::network::constants::Network::Testnet, &seed)?;
+        let master_xpub = ExtendedPubKey::from_private(&secp, &master_xprv);
 
-        // BIP44: m / purpose' / coin_type' / account' / change / address_index
-        // coin_type = 0 bitcoin, 1 testnet, 1776 liquid bitcoin as defined in https://github.com/satoshilabs/slips/blob/master/slip-0044.md
-        // slip44 suggest 1 for every testnet, so we are using it also for regtest
-        let coin_type: u32 = match self.network.id() {
-            NetworkId::Bitcoin(bitcoin_network) => match bitcoin_network {
-                bitcoin::Network::Bitcoin => 0,
-                bitcoin::Network::Testnet
-                | bitcoin::Network::Regtest
-                | bitcoin::Network::Signet => 1,
-            },
-            NetworkId::Elements(elements_network) => match elements_network {
-                ElementsNetwork::Liquid => 1776,
-                ElementsNetwork::ElementsRegtest => 1,
-            },
-        };
+        // TODO: when supporting multi-account, remove derived xprv, xpub
```

Comment:
Hmm, it seems like the current behavior with `xprv`/`xpub` is actually wrong?

They're passed as the master keys to `WalletCtx::new()`, which would result in using a derivation path of `m/49'/0'/0'/<purpose>/0'/<account>` for the subaccounts.

It's also passed to `StoreMeta::new()` where its used as the encryption key, but I guess we need to keep this to retain compatibility, unless we want to also have a `fallback_xpub` argument?

Reply:
> // TODO: when supporting multi-account, remove derived xprv, xpub

I think I should remove the comment, it's kind of misleading


> They're passed as the master keys to WalletCtx::new()

In the current master `WalletCtx::new` expects (xpub, xprv) to be m/49'/0'/0':

    let change_address = self.derive_address(&self.xpub, [1, change_index])?.to_string();

Thus I haven't changed it.
Adding support for (sub)accounts changes the behaviour of `WalletCtx`, but in this PR I'm leaving that as it is.


> It's also passed to StoreMeta::new() where its used as the encryption key

Yes, so we use M/49'/0'/0' (encoded for testnet) to derive the encryption key.
That must be preserved eventually as a fallback to ensure backward compatibility with Aqua.
Some services may require or have required the subaccount xpub for monitoring, so I see the point for changing it.
Do you think it's worth changing it now as well? Should we use the master xpub?

Reply:
Okay, I see, I didn't realize this was going to get merged directly on master.

Is this also going to get released in aqua separately from #93?

Reply:
> Is this also going to get released in aqua separately from #93?

It should not

Reply:
If they're going to get released together, then maybe its worth doing this PR on top of #93?

Reply:
Yes sure we can do that

Reply:
Done